### ISSUE TITLE
[Feedback] - apir-reference feedback from dynaboard

### ISSUE THREAD

**Issue by bruuuuuuuce** [momentohq/public-dev-docs#360]
[slack thread](https://gomomento.slack.com/archives/C05KZ20F5EZ/p1691178983109829)

User reported some confusion trying to get a javascript example started using our `api-reference` docs. 
- they wanted to use an authentication type that was not documented under api reference
- No docs about `topicClient` inside of api-reference
- explain in more detail what the `CredentialsProvider` and `Configurations` classes are, what they do, and what each option is best used for
- explain all constructor options

**Comment by NoSQLKnowHow** [comment]
Some of this stuff is already documented. Specifically, the [credentialprovider and configurations](https://docs.momentohq.com/develop), the customer may not have found it, though.

Also, there are [docs about TopicClient](https://docs.momentohq.com/develop/api-reference/topics) under api-reference. We might need to expand them more though.

Regardless, I am happy to take this.

**Comment by NoSQLKnowHow** [comment]
Got a PR for the docs ready to go that breaks out some of this info into its own pages as well as adding a topics cheat sheet page. I am waiting on the Topics example to be merged into the client-sdk-javascript before this PR can be merged to prod.

**Comment by cprice404** [comment]
I think there is just a little bit of work left to do on this ticket.

Kirk created a "topics cheatsheet" for the JS SDK but (a) it's not linked from the main page yet and (b) the content is quite fleshed out yet. Once we get those two things addressed we can close this ticket.
